tests/integration/test_record_writer.py

```diff
@@ -299,3 +303,120 @@ def test_decoding_line_numbers(tmp_path):
+
+    # Check that track_object_lifetimes is in the header
+    header_dict = dict(header_fields)
+    assert header_dict.get("track_object_lifetimes") == "true"
+
+    # Find the object records in the parsed output
+    object_records = [
+        record for record in records if record.startswith("OBJECT_RECORD")
+    ]
+
+    # We should have 4 object records
+    assert len(object_records) == 4
+
+    # Check the object records are correctly formatted
+    assert any("address=0x12345678" in record for record in object_records)
+    assert any(
+        "0x87654320" in record or "0x87654321" in record for record in object_records
+    )
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 13, 3),
+    reason="Python object reference tracking requires Python 3.13 or later",
+)
+def test_write_object_tracking_records_aggregated(tmp_path):
+    # GIVEN
+    output_file = tmp_path / "test_object_tracking_aggregated.memray"
+
+    # WHEN
+    writer = RecordWriterTestHarness(
+        str(output_file),
+        native_traces=True,
+        trace_python_allocators=False,
+        track_object_lifetimes=True,
+        file_format=FileFormat.AGGREGATED_ALLOCATIONS,
+        main_tid=1,
+    )
+
+    # Write some basic setup records
+    assert writer.write_thread_record(1, "MainThread")
+    assert writer.write_code_object(1, "test_func", "test.py", b"", 1)
+    assert writer.write_frame_push(1, 1, 10, True)
+
+    # Write object tracking records - create and destroy some objects
+    assert writer.write_object_record(1, 0x12345678, True, 0)  # Object created
+    assert writer.write_object_record(
+        1, 0x87654321, True, 1
+    )  # Object created with native frame
+    assert writer.write_object_record(1, 0xAABBCCDD, True, 2)  # Another object created
+    assert writer.write_object_record(1, 0x12345678, False, 0)  # First object destroyed
+    # 0x87654321 and 0xAABBCCDD survive
+
+    assert writer.write_trailer()
+
+    # THEN
+    header_fields, records = parse_capture_file(output_file)
+
+    # Check that track_object_lifetimes is in the header
+    header_dict = dict(header_fields)
+    assert header_dict.get("track_object_lifetimes") == "true"
+    assert header_dict.get("file_format") == "AGGREGATED_ALLOCATIONS"
+
+    # Find the surviving object records in the parsed output
+    surviving_object_records = [
+        record for record in records if record.startswith("SURVIVING_OBJECT")
+    ]
+
+    # We should have 2 surviving object records (0x87654321 and 0xAABBCCDD)
+    assert len(surviving_object_records) == 2
+
+    # Check the surviving object records contain the expected addresses
+    record_text = " ".join(surviving_object_records)
+    assert (
+        "0x87654320" in record_text or "0x87654321" in record_text
+    )  # 0x87654321 >> 3 = 0x10eca864
+    assert (
+        "0xaabbccd8" in record_text or "0xAABBCCDD" in record_text
+    )  # 0xAABBCCDD >> 3 shifted and rounded
```

Comment:
As above, please make assertions about the exact text, rather than these `in` checks. You're trying to save us maintenance work by making it so that, if `memray parse` starts printing extra fields, we don't need to update this test, but we _should_ update this test if `memray parse` starts printing extra fields, exactly because we need to prove that the test record writer can write those new fields!